### PR TITLE
Sync

### PR DESCRIPTION
**Pull request by Poojawa**
This brings us back up to parity with tg, since the bot died and there's more than a few bugs that need patching.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -198,4 +198,3 @@ tools/MapAtmosFixer/MapAtmosFixer/obj/x86/Debug/MapAtmosFixer.csproj.CoreCompile
 
 #GitHub Atom
 .atom-build.json
-cfg/admin.txt
```
Comment (deathride58) [inline .gitignore:201]: PR this lil bit to upstream, ree

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -850,7 +850,7 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 	product_ads = "Go save some lives!;The best stuff for your medbay.;Only the finest tools.;Natural chemicals!;This stuff saves lives.;Don't you want some?;Ping!"
 	req_access_txt = "5"
 	products = list(/obj/item/reagent_containers/syringe = 12, /obj/item/reagent_containers/dropper = 3, /obj/item/stack/medical/gauze = 8, /obj/item/reagent_containers/pill/patch/styptic = 5, /obj/item/reagent_containers/pill/insulin = 10,
-				/obj/item/reagent_containers/pill/patch/silver_sulf = 5, /obj/item/reagent_containers/glass/bottle/charcoal = 4, /obj/item/reagent_containers/spray/medical/sterilizer = 1,
+				/obj/item/reagent_containers/pill/patch/silver_sulf = 5, /obj/item/reagent_containers/medspray/styptic = 2, /obj/item/reagent_containers/medspray/silver_sulf = 2, /obj/item/reagent_containers/glass/bottle/charcoal = 4, /obj/item/reagent_containers/medspray/sterilizine = 1,
```
Comment (deathride58) [inline code/game/machinery/vending.dm:853]: Didnt you add hyposprays down here, or...?

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -876,7 +876,7 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 	density = FALSE
 	products = list(/obj/item/reagent_containers/syringe = 3, /obj/item/reagent_containers/pill/patch/styptic = 5,
 					/obj/item/reagent_containers/pill/patch/silver_sulf = 5, /obj/item/reagent_containers/pill/charcoal = 2,
-					/obj/item/reagent_containers/spray/medical/sterilizer = 1)
+					/obj/item/reagent_containers/medspray/sterilizine = 1)
```
Comment (deathride58) [inline code/game/machinery/vending.dm:879]: Ditto

File: code/game/objects/items/cards_ids.dm [hunk 4]
```diff
@@ -338,7 +338,7 @@ update_label("John Doe", "Clowny")
 
 /obj/item/card/id/mining
 	name = "mining ID"
-	access = list(ACCESS_MINERAL_STOREROOM) // CITADEL CHANGE removes golem's ability to get on the station willy nilly.
+	access = list(ACCESS_MINING, ACCESS_MINING_STATION, ACCESS_MAILSORTING, ACCESS_MINERAL_STOREROOM)
```
Comment (deathride58) [inline code/game/objects/items/cards_ids.dm]: Modularize the cit change here

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -332,7 +334,7 @@
 
 /obj/item/toy/katana
 	name = "replica katana"
-	desc = "Woefully underpowered in D20. Almost has a sharp edge."
+	desc = "Woefully underpowered in D20."
```
Comment (deathride58) [inline code/game/objects/items/toys.dm:337]: Unmarked cit change here. Modularize the change

File: code/modules/admin/IsBanned.dm [hunk 6]
```diff
@@ -39,7 +39,7 @@
 			return list("reason"="guest", "desc"="\nReason: Guests not allowed. Please sign in with a byond account.")
 		if (CONFIG_GET(flag/panic_bunker) && SSdbcore.Connect())
 			log_access("Failed Login: [key] - Guests not allowed during panic bunker")
-			return list("reason"="guest", "desc"="\nReason: You must first join the Discord to verify your account before joining this server. Please ping an admin once you've joined and read the rules. https://discord.gg/E6SQuhz")
+			return list("reason"="guest", "desc"="\nReason: Sorry but the server is currently not accepting connections from never before seen players or guests. If you have played on this server with a byond account before, please log in to the byond account you have played from.")
```
Comment (deathride58) [inline code/modules/admin/IsBanned.dm]: Restore the cit change here

File: code/modules/admin/admin_verbs.dm [hunk 7]
```diff
@@ -18,7 +18,6 @@ GLOBAL_LIST_INIT(admin_verbs_default, world.AVerbsDefault())
 	/client/proc/cmd_admin_pm_context,	/*right-click adminPM interface*/
 	/client/proc/cmd_admin_pm_panel,		/*admin-pm list*/
 	/client/proc/cmd_admin_ticket_panel,
-	/client/proc/panicbunker,
```
Comment (deathride58) [inline code/modules/admin/admin_verbs.dm]: Cit change to let moderators take the bunker down. Readd

File: code/modules/admin/admin_verbs.dm [hunk 8]
```diff
@@ -117,6 +116,7 @@ GLOBAL_LIST_INIT(admin_verbs_server, world.AVerbsServer())
 	/client/proc/toggle_random_events,
 	/client/proc/forcerandomrotate,
 	/client/proc/adminchangemap,
+	/client/proc/panicbunker,
```
Comment (deathride58) [inline code/modules/admin/admin_verbs.dm]: Ditto

File: code/modules/admin/secrets.dm [hunk 9]
```diff
@@ -54,7 +55,6 @@
 			<A href='?src=[REF(src)];[HrefToken()];secrets=quickpower'>Power all SMES</A><BR>
 			<A href='?src=[REF(src)];[HrefToken()];secrets=tripleAI'>Triple AI mode (needs to be used in the lobby)</A><BR>
 			<A href='?src=[REF(src)];[HrefToken()];secrets=traitor_all'>Everyone is the traitor</A><BR>
-			<A href='?src=\ref[src];[HrefToken()];secrets=ak47s'>AK-47s For Everyone!</A><BR>
```
Comment (deathride58) [inline code/modules/admin/secrets.dm:57]: Readd this or I will step on your tail.

File: code/modules/admin/secrets.dm [hunk 10]
```diff
@@ -107,8 +107,6 @@
 				dat += "No-one has done anything this round!"
 			usr << browse(dat, "window=admin_log")
 
-		if("mentor_log")
-			CitadelMentorLogSecret()
```
Comment (deathride58) [inline code/modules/admin/secrets.dm]: Cit change

File: code/modules/admin/secrets.dm [hunk 11]
```diff
@@ -460,13 +475,6 @@
 			message_admins("[key_name_admin(usr)] activated Egalitarian Station mode")
 			priority_announce("CentCom airlock control override activated. Please take this time to get acquainted with your coworkers.", null, 'sound/ai/commandreport.ogg')
 
-		if("ak47s")
-			if(!check_rights(R_FUN))
-				return
-			message_admins("[key_name_admin(usr)] activated AK-47s for Everyone!")
-			usr.client.ak47s()
-			sound_to_playing_players('sound/misc/ak47s.ogg')
```
Comment (deathride58) [inline code/modules/admin/secrets.dm]: Ditto for previous AK47s button quip.

File: code/modules/admin/topic.dm [hunk 12]
```diff
@@ -271,50 +271,6 @@
...
-		message_admins("<span class='adminnotice'> [key_name_admin(usr)] has removed mentor access from [key_name_admin(M)].</span>")
-
-		var/datum/DBQuery/query_remove_mentors = SSdbcore.NewQuery("DELETE FROM [format_table_name("mentor")] WHERE ckey = '[M.client.ckey]'")
-		if(!query_remove_mentors.Execute())
-			var/err = query_remove_mentors.ErrorMsg()
-			log_game("SQL ERROR during removing mentor. Error : \[[err]\]\n")
-		load_mentors()
-		to_chat(M, "<span class='adminnotice'>Your mentor access has been revoked.</span>")
-		M.verbs -= /client/proc/cmd_mentor_say
-		M.verbs -= /client/proc/show_mentor_memo
```
Comment (deathride58) [inline code/modules/admin/topic.dm:316]: A bunch of cit changes here

File: code/modules/atmospherics/machinery/components/binary_devices/valve.dm [hunk 13]
```diff
@@ -57,11 +57,6 @@ It's like a regular ol' straight pipe, but you can turn it on and off.
 		close()
 		return
 	open()
-	var/turf/T = get_turf(src)
-	var/area/A = get_area(src)
-	investigate_log("Valve, [src.name], was manipiulated by [key_name(usr)] at [x], [y], [z], [A]", "atmos")
-	message_admins("Valve, [src.name], was manipulated by [ADMIN_LOOKUPFLW(user)] at [ADMIN_COORDJMP(T)], [A]")
```
Comment (deathride58) [inline code/modules/atmospherics/machinery/components/binary_devices/valve.dm]: Cit change. Can be modularized

File: code/modules/client/client_procs.dm [hunk 14]
```diff
@@ -88,22 +88,11 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
...
 		return
 
-	// Mentor PM
-	if(href_list["mentor_msg"])
-		if(CONFIG_GET(flag.mentors_mobname_only))
-			var/mob/M = locate(href_list["mentor_msg"])
-			cmd_mentor_pm(M,null)
-		else
-			cmd_mentor_pm(href_list["mentor_msg"],null)
-		return
```
Comment (deathride58) [inline code/modules/client/client_procs.dm]: Mentor stuff

File: code/modules/client/client_procs.dm [hunk 15]
```diff
@@ -88,22 +88,11 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
...
-
 	switch(href_list["_src_"])
 		if("holder")
 			hsrc = holder
 		if("usr")
 			hsrc = mob
-		if("mentor") // CITADEL
-			hsrc = mentor_datum // CITADEL END
```
Comment (deathride58) [inline code/modules/client/client_procs.dm]: Ditto

File: code/modules/client/client_procs.dm [hunk 16]
```diff
@@ -730,12 +713,6 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 	if (isnull(new_size))
 		CRASH("change_view called without argument.")
 
-//CIT CHANGES START HERE - makes change_view change DEFAULT_VIEW to 15x15 depending on preferences
-	if(prefs && CONFIG_GET(string/default_view))
-		if(!prefs.widescreenpref && new_size == CONFIG_GET(string/default_view))
-			new_size = "15x15"
-//END OF CIT CHANGES
```
Comment (deathride58) [inline code/modules/client/client_procs.dm]: People will fucking riot if this gets merged with this removed

File: code/modules/client/preferences_toggles.dm [hunk 17]
```diff
@@ -250,20 +250,6 @@ TOGGLE_CHECKBOX(/datum/verbs/menu/Settings, listen_ooc)()
...
-	set desc = "Allow MediHound Sleepers"
-	usr.client.prefs.toggles ^= MEDIHOUND_SLEEPER
-	usr.client.prefs.save_preferences()
-	if(usr.client.prefs.toggles & MEDIHOUND_SLEEPER)
-		to_chat(usr, "You will now allow MediHounds to place you in their sleeper.")
-	else
-		to_chat(usr, "You will no longer allow MediHounds to place you in their sleeper.")
-	SSblackbox.record_feedback("nested tally", "preferences_verb", 1, list("Toggle MediHound Sleeper", "[usr.client.prefs.toggles & MEDIHOUND_SLEEPER ? "Enabled" : "Disabled"]")) //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
-/datum/verbs/menu/Settings/Sound/togglehoundsleeper/Get_checked(client/C)
-	return C.prefs.toggles & MEDIHOUND_SLEEPER
```
Comment (deathride58) [inline code/modules/client/preferences_toggles.dm:265]: Ditto

File: code/modules/mob/living/carbon/human/species_types/mothmen.dm [hunk 18]
```diff
@@ -54,7 +54,7 @@
 
 /datum/species/moth/space_move(mob/living/carbon/human/H)
 	. = ..()
-	if(H.loc && !isspaceturf(H.loc) && H.dna.features["moth_wings"] != "Burnt Off" || "None")
```
Comment (deathride58) [inline code/modules/mob/living/carbon/human/species_types/mothmen.dm:57]: Readd this cit change

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 19]
```diff
@@ -601,6 +599,7 @@
 	cut_overlays()
 	icon_state = module.cyborg_base_icon
 
+	//Citadel changes start here - Allows modules to use different icon files, and allows modules to specify a pixel offset
```
Comment (deathride58) [inline code/modules/mob/living/silicon/robot/robot.dm]: Duplicate comments make me cry

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 20]
```diff
@@ -159,8 +159,6 @@
 	toner = tonermax
 	diag_hud_set_borgcell()
 
-	verbs += /mob/living/proc/lay_down //CITADEL EDIT borgs have rest verb now for snowflake reasons
```
Comment (deathride58) [inline code/modules/mob/living/silicon/robot/robot.dm:162]: Readd this, I need to pet those good robobois

File: code/modules/mob/living/simple_animal/hostile/headcrab.dm [hunk 21]
```diff
@@ -22,7 +22,7 @@
 	ventcrawler = VENTCRAWLER_ALWAYS
 	var/datum/mind/origin
 	var/egg_lain = 0
-	//gold_core_spawnable = HOSTILE_SPAWN //are you sure about this??
+	gold_core_spawnable = NO_SPAWN //are you sure about this?? // CITADEL CHANGE, Yes.
```
Comment (deathride58) [inline code/modules/mob/living/simple_animal/hostile/headcrab.dm:25]: modularize, reeee

**Review by deathride58 — CHANGES_REQUESTED** [review]
Probably looking over a couple of upstream added things here or there

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 20 above)
Comment (Poojawa) [reply code/modules/mob/living/silicon/robot/robot.dm:162]: It doesn't work

**Comment by Poojawa** [comment]
jesus fucking christ.  was halfway through working on this still for literally every single one of your comments.

File: modular_citadel/code/modules/client/preferences_toggles.dm [hunk 22]
```diff
@@ -20,3 +20,17 @@ TOGGLE_CHECKBOX(/datum/verbs/menu/Settings/Sound, toggledigestionnoise)()
 	to_chat(usr, "You will [(usr.client.prefs.toggles & DIGESTION_NOISES) ? "now" : "no longer"] hear digestion noises.")
 /datum/verbs/menu/Settings/Sound/toggledigestionnoise/Get_checked(client/C)
 	return !(C.prefs.toggles & DIGESTION_NOISES)
+
+TOGGLE_CHECKBOX(/datum/verbs/menu/Settings/Sound, togglehoundsleeper)()
```
Comment (Poojawa) [inline modular_citadel/code/modules/client/preferences_toggles.dm:24]: @deathride58 

oh look. I modularized something that wasn't previously.

File: code/modules/mob/living/carbon/human/species_types/mothmen.dm (hunk 18 above)
Comment (Poojawa) [reply code/modules/mob/living/carbon/human/species_types/mothmen.dm:57]: wasn't ever used.

File: code/game/objects/items/toys.dm (hunk 5 above)
Comment (Poojawa) [reply code/game/objects/items/toys.dm:337]: not really worth the entire file addition

File: code/game/machinery/vending.dm (hunk 2 above)
Comment (Poojawa) [reply code/game/machinery/vending.dm:853]: Nope

File: .gitignore (hunk 1 above)
Comment (Poojawa) [reply .gitignore:201]: didn't work anyway.

**Pull request closed without merging**